runconfig/hostconfig_windows.go

```diff
@@ -71,3 +70,15 @@ func ValidateResources(hc *container.HostConfig, si *sysinfo.SysInfo) error {
 	}
 	return nil
 }
+
+// ValidatePrivileged performs platform specific validation of the Privileged setting
+func ValidatePrivileged(hc *container.HostConfig) error {
+	// We may not be passed a host config, such as in the case of docker commit
+	if hc == nil {
+		return nil
+	}
+	if hc.Privileged {
+		return fmt.Errorf("invalid --privileged: Windows does not support this feature")
```

Comment:
That's ok for now; agreed on consistency. Given that the API can be used separate from the docker-cli, we should probably update them all at some point.

@jhowardmsft perhaps you can review https://github.com/docker/docker/pull/30788 as well, which handles these things on the client?